### DISCUSSION TITLE
Question about usePaginatedQuery

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#1128] [General]
In my application, I have a data table has pagination. Initially, I was using useQuery, but the data was undefined every time the page changes. So I switched to usePaginatedQuery. Now it works like I expected it to work. The resolvedData has the old data until the new page is fetched. However, isLoading is always false. I still want to know if the data is being fetched, so I can indicate that a progress, without clearing the data. 

Is there a way to get the loading state for usePaginatedQuery?

Also, I saw somewhere in the forum that usePaginatedQuery will not be available in 3.x. Will there be an equivalent feature?

Thanks
Anand

**Comment by nanandn** [comment]
Looks like I can use isFetching state.

**Reply by TkDodo** [reply]
Yes, isLoading is only true for the „initial“ loading, while isFetching is true every time a request is in-flight, which includes the first load as well as background refetches.

> Also, I saw somewhere in the forum that usePaginatedQuery will not be available in 3.x. Will there be an equivalent feature?

Since the [implementation ](https://github.com/tannerlinsley/react-query/blob/675605024d96aa99989cbeba2192b70f62be53da/src/react/usePaginatedQuery.ts#L63) of `usePaginatedQuery` is quite simple (based on `keepPreviousData` flag), I think there will still be a way to make it work in v3, but I don’t know the plans.

**Reply by LeulAria** [reply]
ya same issue here
`status==="loading"` and `isLoading` are same for paginate query or refetch 
and  `isFetching` works 4me...

**Comment by dayoolacodes** [comment]
thanks [nanandn](https://github.com/nanandn)
